pkg/reconciler/v1alpha1/pipelinerun/resources/pipelinerunresolution.go

```diff
@@ -218,6 +224,22 @@ func GetPipelineConditionStatus(prName string, state []*ResolvedPipelineRunTask,
 		}
 	}
 	if !allFinished {
+		if !startTime.IsZero() && pipelineTimeout != nil {
+			timeout := pipelineTimeout.Duration
+			runtime := time.Since(startTime.Time)
+			if runtime > timeout {
+				logger.Infof("PipelineRun %q has timed out(runtime %s over %s)", prName, runtime, timeout)
+
+				timeoutMsg := fmt.Sprintf("PipelineRun %q failed to finish within %q", prName, timeout.String())
+				return &duckv1alpha1.Condition{
+					Type:    duckv1alpha1.ConditionSucceeded,
+					Status:  corev1.ConditionFalse,
+					Reason:  ReasonTimedOut,
+					Message: timeoutMsg,
+				}
```

Comment:
after this happens, does anything need to be done to make sure the taskruns that the pipelinerun spawned will also time out?

Reply:
Hum yeah I think we need to cancel the `TaskRun`s there :angel:

Reply:
The `TaskRun`s should themselves time out now.

Reply:
@abayer oh right, `TaskRun`s will also have the timeout field, and thus should time out on their own in the same time more or less, right ?

Reply:
But I don't see the code for that though :thinking: Am I missing something ?

Reply:
@vdemeester See the changes to`createTaskRun` in https://github.com/knative/build-pipeline/pull/355/files#diff-225ed18b17b573cba99d592dea1da1a0R308

Reply:
@abayer nevermind, I was looking in the diff for support for `Timeout` on Task.… but it's already supported (because of build) :sweat_smile: